### ISSUE TITLE
Banners Lose Patterns

### ISSUE THREAD

**Issue by WenXin20** [Direwolf20-MC/BuildingGadgets#538]
`Minecraft v1.15.2`
`Forge v31.2.45`
`Building Gadgets v3.4.0`

Banners placed by the copy-paste gadget lose their patterns.

**Comment by MichaelHillcox** [comment]
Yeah, I think this is an issue with the botched TileEntity support, I’ll have a look into it :)

**Comment by WenXin20** [comment]
Thanks! Much appreciated.

**Comment by MichaelHillcox** [comment]
I'm going to fix this on 1.16 and backport it once it's done, It'll miss 3.4.1 unfortunatly as it's a big fix

**Comment by hvdklauw** [comment]
Ok, did some debugging, the issue is with the copy part of the gadget.

On [InventoryHelper.java#L279](https://github.com/Direwolf20-MC/BuildingGadgets/blob/11c42c93e1ab4a1c12518e1ccf798c5d5243a246/src/main/java/com/direwolf20/buildinggadgets/common/tainted/inventory/InventoryHelper.java#L279) the result of the `TileSupport.createTileData(world, pos)` is null.

Diving down we get to [TileSupport.java#L39](https://github.com/Direwolf20-MC/BuildingGadgets/blob/11c42c93e1ab4a1c12518e1ccf798c5d5243a246/src/main/java/com/direwolf20/buildinggadgets/common/tainted/building/tilesupport/TileSupport.java#L39) which loops just once and `res = null`

On line [TileSupport.java#L66](https://github.com/Direwolf20-MC/BuildingGadgets/blob/11c42c93e1ab4a1c12518e1ccf798c5d5243a246/src/main/java/com/direwolf20/buildinggadgets/common/tainted/building/tilesupport/TileSupport.java#L66) we check if the entity is an instance of `ITileDataProvider` which it isn't because it's a minecraft banner.

Somewhere in there it has to get the correct data from the entity, but I'm not sure where; also because certain other items shouldn't be restorable with all their data (chest contents, or other dupe bugs)
